fdb-extensions/src/main/java/com/apple/foundationdb/async/MoreAsyncUtil.java

```diff
@@ -59,9 +59,14 @@ public class MoreAsyncUtil {
     private static ScheduledThreadPoolExecutor scheduledThreadPoolExecutor
             = new ScheduledThreadPoolExecutor(1, new ThreadFactoryBuilder().setDaemon(true).build());
 
+    public static final CompletableFuture<Boolean> ALREADY_CANCELLED;
```

Comment:
I think the thing that feels weird about this (when it's in the context of `MoreAsyncUtil`) is that you could imagine wanting to have different generic values of future, sort of like `Optional.empty()`. In theory, you can do it by making `ALREADY_CANCELLED` `private` and of type `CompletableFuture<?>`, and then introducing a method to cast it to the right generic type:

```java
    @SuppressWarnings("unchecked")
    public static <T> CompletableFuture<T> alreadyCancelled() {
        return (CompletableFuture<T>)ALREADY_CANCELLED;
    }
```

It does rely on type erasure and also the fact that we know that the cancelled future will never actually return a `T`, but it does make it a bit more versatile.

Reply:
That is fair. I don't like the idea of using type erasure to do this.
Instead I created that method, and then `FlatMapPipelinedCursor` and `MapPipelinedCursor` put the `Boolean` version in a constant.